homeassistant/components/rabbitair/entity.py

```diff
@@ -0,0 +1,75 @@
+"""A base class for Rabbit Air entities."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from rabbitair import Client, Model, State
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+MODELS = {
+    Model.A3: "A3",
+    Model.BioGS: "BioGS 2.0",
+    Model.MinusA2: "MinusA2",
+    None: None,
+}
+
+
+class RabbitAirBaseEntity(CoordinatorEntity[State]):
+    """Base class for Rabbit Air entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[State],
+        client: Client,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._client = client
+        self._entry = entry
+        self._attr_unique_id = entry.unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, entry.unique_id)},  # type: ignore[arg-type]
+            manufacturer="Rabbit Air",
+            model=MODELS.get(coordinator.data.model),
+            name=entry.title,
+            sw_version=coordinator.data.wifi_firmware,
+            hw_version=coordinator.data.main_firmware,
+        )
+
+    def _is_model(self, model: Model | list[Model]) -> bool:
+        """Check the model of the device."""
+        if isinstance(model, list):
+            return self.coordinator.data.model in model
+        return self.coordinator.data.model is model
+
+    async def _set_state(self, **kwargs: Any) -> None:
+        """Change the state of the device."""
+        _LOGGER.debug("Set state %s", kwargs)
+        await self._client.set_state(**kwargs)
+        # Wait for the changes to be applied and for the device status to be
+        # updated. Two seconds should be sufficient, since the internal cycle
+        # of the device runs at one-second intervals.
+        await asyncio.sleep(2)
+        # Force polling of the device, because changing one parameter often
+        # causes other parameters to change as well. By getting updated status
+        # immediately we provide a better user experience, especially if the
+        # default polling interval is set too long.
+        await self.coordinator.async_refresh()
```

Comment:
Once the cooldown/immediate update has been adjusted on the coordinator, you can adjust this:

```suggestion
        await self.coordinator.async_request_refresh()
```

Reply:
This turned out to be a little more complicated. When I just set cooldown=2 and immediate=false, there was a weird effect in the user interface: after changing the state of the controls, they could go back to their previous state. It turned out that the effect is composed of two parts.
1. There was the possibility of polling the device before 2 seconds had elapsed and receiving a still unchanged state. This can happen either because of periodic polling (it happens immediately at the scheduled time, and cancels the cooldown delay), or because several requests were sent during the cooldown period (since the delay time does not start counting again).
2. There seems to be a 2-second timeout somewhere in the user interface. This is especially noticeable on the power switch. If the state reported by the entity has not changed within this time, the switch returns to its previous state. But then, almost immediately, a new state comes from the device, and the switch jumps back again.

I made some changes to avoid this effect. Could you take a look? Maybe there is some other, more correct, solution?